File: sdks/php/examples/list_products.php
<?php

require_once __DIR__ . '/../vendor/autoload.php';

use Puxbay\Puxbay;
use Puxbay\PuxbayConfig;

$apiKey = getenv('PUXBAY_API_KEY');
if (!$apiKey) {
    echo "PUXBAY_API_KEY must be set\n";
    exit(1);
}

$config = new PuxbayConfig($apiKey);
$client = new Puxbay($config);

echo "Fetching products...\n";

try {
    $products = $client->products->list(1);
    foreach ($products->results as $product) {
        echo "- {$product->name} (\${$product->price})\n";
    }
} catch (Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
}

File: sdks/dart/example/list_products.dart
import 'dart:io';
import 'package:puxbay/puxbay.dart';

void main() async {
  final apiKey = Platform.environment['PUXBAY_API_KEY'];
  if (apiKey == null) {
    print('PUXBAY_API_KEY must be set');
    exit(1);
  }

  final config = PuxbayConfig();
  final client = Puxbay(apiKey: apiKey, config: config);

  print('Fetching products...');
  try {
    final products = await client.products.list(page: 1);
    for (var product in products.results) {
      print('- ${product.name} (\$${product.price})');
    }
  } catch (e) {
    print('Error: $e');
  }
}
